app/controllers/application_controller.rb

```diff
@@ -109,4 +109,10 @@ def after_sign_in_path_for(_)
     dashboard_path
   end
 
+  # raises error if match against regular expression patterns fails
+  def check_redirect(url)
+    redirect_to url if URI.parse(url).path
+    rescue URI::InvalidURIError
+      redirect_to '/'
```

Comment:
`rescue` will be triggered only when someone has sent something is params which redirects to something other than our site. This doesn't solves the issue that user can't send something dubious as params, it only makes sure that those malicious links don't work, which in turn solves the issue that user won't send anything malicious as those efforts will be futile. I hope you understand.